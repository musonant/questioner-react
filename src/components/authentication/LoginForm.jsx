import React, { Component } from 'react';
import SimpleReactValidator from 'simple-react-validator';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { loginUser } from '../../store/modules/auth';

class LoginForm extends Component {
  state = {
    email: '',
    password: '',
  };

  validator = new SimpleReactValidator({
    element: message => <div>{message}</div>,
  });

  onChange = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  formSubmitHandler = async e => {
    e.preventDefault();
    this.validator.showMessages();
    this.setState({
      ...this.state,
    });

    if (!this.validator.allValid()) {
      return false;
    }

    const credentials = {
      email: this.state.email,
      password: this.state.password,
    };

    await this.props.loginUser(credentials);

    if (this.props.auth.currentUser.id) {
      this.redirect();
    }
  };

  redirect = () => {
    this.props.history.push('/');
  };

  render() {
    const { loginStatus } = this.props;

    return (
      <form
        onSubmit={this.formSubmitHandler}
        id="signin-form"
        className={`form ${loginStatus}`}
      >
        <input
          type="text"
          name="email"
          placeholder="email"
          value={this.state.email}
          onChange={this.onChange}
        />
        <input
          type="password"
          name="password"
          value={this.state.password}
          placeholder="password"
          onChange={this.onChange}
        />
        <div className="err-msg">
          {this.validator.message('email', this.state.email, 'required|email')}
          {this.validator.message(
            'password',
            this.state.password,
            'required|min:6|alpha_num',
          )}
        </div>
        <input type="submit" value="Login" />
      </form>
    );
  }
}

LoginForm.propTypes = {
  loginStatus: PropTypes.string.isRequired,
  loginUser: PropTypes.func,
  errorMessage: PropTypes.string,
  history: PropTypes.object.isRequired,
  auth: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  auth: state.auth,
  errorMessage: state.auth.errorMessage,
});

export default connect(
  mapStateToProps,
  { loginUser },
)(LoginForm);
