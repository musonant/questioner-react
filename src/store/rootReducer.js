import { combineReducers } from 'redux';
import { authReducer } from './modules/auth';
import { meetupReducer } from './modules/meetup';

export default combineReducers({
  auth: authReducer,
  meetups: meetupReducer
});
