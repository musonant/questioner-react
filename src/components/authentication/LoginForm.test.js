import React from 'react';
import { shallow } from 'enzyme';

import store from '../../store/';
import LoginForm from './LoginForm';

const props = {
  auth: {
    isLoading: false,
    errorMessage: '',
    currentUser: {}
  },
  history: {},
  loginStatus: 'login',
  loginUser: jest.fn()
};

describe('Login Form', () => {
  it('Should render correctly in debug mode', () => {
    const component = shallow(<LoginForm store={store} {...props} />);
    expect(component).toMatchSnapshot();
  });
});
