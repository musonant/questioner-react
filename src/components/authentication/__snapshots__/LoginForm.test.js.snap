// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Should render correctly in debug mode 1`] = `
<ContextProvider
  value={null}
>
  <LoginForm
    auth={
      Object {
        "currentUser": Object {},
        "errorMessage": "",
        "isLoading": false,
      }
    }
    errorMessage=""
    history={Object {}}
    loginStatus="login"
    loginUser={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
